typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -113,8 +123,102 @@ export class EvmHookModule extends HyperlaneModule<
       : this.reader.deriveHookConfig(this.args.addresses.deployedHook);
   }
 
-  public async update(_config: HookConfig): Promise<AnnotatedEV5Transaction[]> {
-    throw new Error('Method not implemented.');
+  public async update(
+    targetConfig: HookConfig,
+  ): Promise<AnnotatedEV5Transaction[]> {
+    targetConfig = HookConfigSchema.parse(targetConfig);
+
+    // Do not support updating to a custom Hook address
+    if (typeof targetConfig === 'string') {
+      throw new Error(
+        'Invalid targetConfig: Updating to a custom Hook address is not supported. Please provide a valid Hook configuration.',
+      );
+    }
+
+    // save current config for comparison
+    // normalize the config to ensure it's in a consistent format for comparison
+    const currentConfig = normalizeConfig(await this.read());
+
+    // Update the config
+    this.args.config = targetConfig;
+
+    // If configs match, no updates needed
+    if (configDeepEquals(currentConfig, targetConfig)) {
+      return [];
+    }
```

Comment:
thoughts on declaring `let updateTxs: AnnotatedEV5Transaction[];` early on, and returning this instead of `[]`?

nit: remove comment and inline like `if (...) return updateTxs;`

Reply:
I usually prefer to `return []` to make it explicit what should happen

Reply:
re inline if statements, i'm not really a fan of them at all but will adopt if it makes its way into the linter

keeping as-is because there's probably a bunch of other places that exist where we have a mix of inline and normal